gnd/src/main/java/com/google/android/gnd/model/basemap/Area.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gnd.model.basemap;
+
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.auto.value.AutoValue;
+
+/**
+ * An area is a contiguous set of tiles that form a geodesic rectangle.
+ * They are uniquely identifiable by their NE and SW bounds (or NW and SE).
+ */
+@AutoValue
+public abstract class Area {
+    public enum State {
+        PENDING,
+        IN_PROGRESS,
+        DOWNLOADED,
+        FAILED
+    }
+
+    public abstract String getId();
+
+    public abstract State getState();
+
+    public abstract LatLngBounds getBounds();
+
+    public static Builder newBuilder() {
+        return new AutoValue_Area.Builder();
+    }
+
+    public abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+        abstract LatLngBounds getBounds();
+
+        public abstract Builder setBounds(LatLngBounds bounds);
+
+        public abstract Builder setState(State state);
+
+        abstract Builder setId(String id);
+
+        abstract Area autoBuild();
+
+        public Area build() {
+            setId(getBounds().toString());
```

Comment:
Rather than hard code this into the builder, it'd be simpler and more consistent to let the caller call setId(), generating a UUID using our UUidGenerator class.

Reply:
A much better approach, thanks! 

We'll expose setId and handle generating a uuid in the datarepo. Since the bounds are no longer used as an identifier, I've added an additional field to the entity to persist them.